src/mon/MonClient.cc

```diff
@@ -1368,3 +1271,93 @@ void MonClient::register_config_callback(md_config_t::config_callback fn) {
 md_config_t::config_callback MonClient::get_config_callback() {
   return config_cb;
 }
+
+class monc_error_category : public ceph::converting_category {
+public:
+  monc_error_category(){}
+  const char* name() const noexcept override;
+  std::string message(int ev) const noexcept override;
+  boost::system::error_condition default_error_condition(int ev) const noexcept
+    override;
+  bool equivalent(int ev, const boost::system::error_condition& c) const
+    noexcept override;
+  using ceph::converting_category::equivalent;
+  int from_code(int ev) const noexcept override;
+};
+
+const char* monc_error_category::name() const noexcept {
+  return "monc";
+}
+
+std::string monc_error_category::message(int ev) const noexcept {
+  using namespace ::monc_errc;
+
+
+  switch (ev) {
+  case 0:
```

Comment:
Discomfort: an error code that's not an error. But I can't think of a better idea right now. :-|

Reply:
That's just a feature of boost::sysem::error_code/std::error_code.

(And boost::system::error_code will be adding MORE error codes that are successes. Or at least they're planning to. Which I have some discomfort with. I'd rather use std::expected. Though it does jive with the desire to be able to, say, have an error category mapping all HTTP error codes to error_code objects.)